diff --git a/bin/perlcalc b/bin/perlcalc
index 657cb0cc..624d8293 100755
--- a/bin/perlcalc
+++ b/bin/perlcalc
@@ -36,8 +36,17 @@ if ($files) {
     magic_filehandle();
 } else {
     if (scalar @ARGV) {
-        foreach my $arg (@ARGV) {
-            say result_of $arg;
+        foreach (@ARGV) {
+            if ($curly_brackets_only) {
+                curly_brackets();
+            } elsif (m{[\{\}]}) {
+                # If any {...} expressions are found, replace them.
+                curly_brackets();
+            } else {
+                # Treat the whole line as an expression.
+                whole_line();
+            }
+            print "\n";
         }
     } else {
         magic_filehandle();
@@ -67,7 +76,9 @@ sub magic_filehandle {
         if (defined $first_line) {
             # If we've reached this point, there's only one line.
             for ($first_line) {
-                if (m{[\{\}]}) {
+                if ($curly_brackets_only) {
+                    curly_brackets();
+                } elsif (m{[\{\}]}) {
                     # If any {...} expressions are found, replace them.
                     curly_brackets();
                 } else {
@@ -143,6 +154,12 @@ standard input to not be read.
     26400
     8.04672
 
+Each argument containing a pair of curly braces has its
+curly-bracketed subexpressions evaluated:
+
+    $ perlcalc '5 miles is {5 * 5280} feet, or {5 * 1.609344} kilometers.'
+    5 miles is 26400 feet, or 8.04672 kilometers.
+
 Use the C<--files> option to treat any non-option argument(s) as
 filename(s) instead of expression(s):
 
